lib/webauth.js

```diff
@@ -246,8 +246,26 @@ module.exports = function(authManager) {
         return;
       }
       const authPluginSession = getAuthPluginSession(req1, handler.pluginID, {});
-      handler.addProxyAuthorizations(req1, req2Options, authPluginSession);     
+      try {
+        handler.addProxyAuthorizations(req1, req2Options, authPluginSession);
+      } catch (e) {
+        authLogger.warn(`Failed to set proxy authorizations. Error=`,e);
+      }
     },
+
+    processProxiedHeaders(req, headers) {
+      const handler = getAuthHandler(req, authManager);
+      if (handler && handler.processProxiedHeaders) {
+        const authPluginSession = getAuthPluginSession(req, handler.pluginID, {});
+        try {
+          return handler.processProxiedHeaders(req, headers, authPluginSession);
```

Comment:
If someone were to use something other than zss-auth, the bug would still be present since the headers are returned in the case of an exception.  Should this function perhaps also be added to zosmf-auth as well?

Reply:
Possibly, yes. Need to check on that next.